book.md

```diff
@@ -1490,6 +1490,169 @@ now explore what we can do about these.
+result can be affected by other factors. For example, in normal environments 
+there are many other source of RF noises that affects the accuracy of signal 
+measurement. In a “noisy” environment, the RF signals from various wireless 
+communication gadgets nearby might drown out the signals from the device being 
+monitored.
+
+### Side channel leakage at instruction level
+Instruction executions can result in various forms of side channel leakage:
+
+Cycle timing resulting from conditional branch – A code sequence containing a 
+conditional branch could result in observable side channel leakage. For example, 
+if the power signatures of several code segments are easily recognizable 
+(process A, B and C in the following diagram), it is possible to detect if the 
+conditional branch was taken or not.
+
+
+![leakage of conditional branch](img/side_channel_leakage_figure_1_branch){ width=80% }
+
+Cycle timing resulting from specific data values – The execution cycles of some 
+instructions can be dependent on the values of input data, resulting in timing 
+side-channel leakage. E.g., the integer divide instruction in Arm Cortex-M 
+processors.
+
+![leakage of execution cycle](img/side_channel_leakage_figure_2_duration){ width=80% }
+
+Power variation due to value changes – The   power spikes in the power signature 
+are often dependent of how many bits are toggled in the register(s), so the 
+amplitude of the spike could be used to tell how many register bits has changed 
+in that clock cycle. The power spikes can be caused by a combination of
+- Logic switching due to the operations of an instruction (e.g. power consumed by a single cycle multiplier can be much higher than the power used by a Boolean logic function), and
+- Logic switching due to changes in data values in the register bank and data paths.
+
+The switching activities are dependent on preceding and next operations. If the 
+power signature of the codes around a specific instruction is recognizable, then 
+the data value being process could be guessed.
+
+![leakage of number of bit toggled](img/side_channel_leakage_figure_3_toggle){ width=80% }
+
+In some SoC or microcontroller implementations, the power spike effect of the 
+operations can be much higher than the effect of data value changes in the 
+register banks.  In such case the program execution flow can be observed, and 
+as a result, might also indirectly leak information about the data that it is 
+processing.
+
+### Countermeasures
+For normal embedded devices that don’t have physical protection features, there 
+is a much higher chance that power/voltage/radiation side channels can result 
+in information leakage. However, some aspects of timing signature leakage could 
+be reduced:
+- Using data processing instruction with data independent timing for cryptographic operations. In recent Arm architectures (including Armv8-A and Armv8-M), some instructions are architecturally defined as DIT (Data Independent Timing).
```

Comment:
This also needs a blank line before the start of the bullet point list.

Reply:
Blank lines added.